src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -1006,7 +1006,7 @@ private bool TryOpen(TaskCompletionSource<DbConnectionInternal> retry)
             }
 
             if (StatisticsEnabled ||
-                s_diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
+                (s_diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand) && Statistics != null ))
```

Comment:
Nit: it's a little strange that ```Statistics``` is used here but ```_statistics``` is used below.  I'd find it easier to understand if both were either ```Statistics``` or ```_statistics```, so that it was clear we're only doing the assignment if the thing we're assigning isn't null.